### DISCUSSION TITLE
Zustand State Access: Destructuring vs. Function Definition

### DISCUSSION THREAD

**Discussion by upromeer** [pmndrs/zustand#2388] [Q&A] [answered]
Is there any difference or potential inconsistency between the following two approaches when dealing with Zustand state in multiple components?

1. Using state destructuring:

   ```jsx
   const [lists, index] = useStore((state) => [state.lists, state.index]);
   const list = lists[index];
   ```

2. Defining a 'list' function inside the store

   ```jsx
   const [lists, index] = useStore((state) => [state.lists, state.index]);
   const list = lists[index];
   ```
   
### Store
   ```js
       (set, get) => {
                return {
                    lists: [],
                    index: 0,
                    list: () => get().lists[get().index],
                    setIndex: (index: number) => set({ index }),
                };
         }
   ```

also partializing the function of store having any problem?

**Comment by dbritto-dev** [comment, marked as answer]
@upromeer let's say you use `equalityFn` then the list function would be the same always and wouldn't re-render I'll would use the first approach

**Reply by upromeer** [reply]
sorry, i made mistake in 2 approach.
have a look!
```js
const list = useStore((state) => state.list);
const foo = list()
```

**Reply by dbritto-dev** [reply]
@upromeer same answer

**Reply by upromeer** [reply]
thanks man
